### ISSUE TITLE
Add package oriented plugin loading support

### ISSUE THREAD

**Issue by MythicManiac** [molenzwiebel/Wetstone#1]
The current way of loading wetstone plugins is not easily manageable in a package oriented fashion, given that it expects a flat structure. This is especially true for automated mod managers which are likely to require manual hardcoded rules and conflict resolution logic, on top of being less reliable.

It'd be great if the plugin loading supported subdirectories, allowing each package to have their own directory, and a `WetstonePlugins` direcotry within that. This would make it a bit easier already as long as the mod managers add hardcoded rules for handling the wetstone plugin content type.

An even better solution would be to automatically discover plugins from `BepInEx/Plugins/*/WetstonePlugins` directories, as bepinex already supports that paradigm and it's widely used.

I took a brief look at the code and it looks like this should be relatively simple, just a matter of changing the plugin discovery logic a bit. If the intention is to keep it configurable, one could add at least partial regex support, and perhaps allow lists in the config value.

So to summarize, I suggest adding support for loading wetstone plugins from `BepInEx/Plugins/*/WetstonePlugins` exclusively or additionally to the current load path. This will make mod management in a package oriented fashion without hardcoded rules for wetstone possible.

**Comment by MythicManiac** [comment]
As discussed, this isn't actually relevant as the `WetstonePlugins` folder is intended for dev use only (hotreloaded plugins) and prod installs / end user installs are going to reside in the normal plugins dir.

So as long as that remains true, there's no need to complicate the reload feature. Closing as solved
